src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -2096,6 +2138,14 @@ private string ExecuteMSBuildExeExpectSuccess(string projectContents, IDictionar
                     }
                 }
 
+                if (envsToCreate != null)
+                {
+                    foreach (var env in envsToCreate)
+                    {
+                        testEnvironment.SetEnvironmentVariable(env.Key, env.Value);
```

Comment:
Will this affect other tests run concurrently?

Reply:
Looks like it probably will..

Reply:
However I got this pattern from a bunch of existing tests.. so I'm not sure what the parallelism story is for the msbuild unit tests..

Reply:
No concurrency among our tests, this is ok. Or at least "not worse than before".